moveit_ros/visualization/planning_scene_rviz_plugin/include/moveit/planning_scene_rviz_plugin/planning_scene_display.h

```diff
@@ -122,16 +122,15 @@ protected Q_SLOTS:
 protected:
   /// This function reloads the robot model and reinitializes the PlanningSceneMonitor
   /// It can be called either from the Main Loop or from a Background thread
-  void loadRobotModel(const rclcpp::Node::SharedPtr& node);
```

Comment:
You should already have access to a node through the Display parent class's `context_` member. 

Taken from `InteractiveMarkerDisplay.cpp`:

```
  auto ros_node_abstraction = context_->getRosNodeAbstraction().lock();
  if (!ros_node_abstraction) {
    return;
  }

  interactive_marker_namespace_property_->initialize(ros_node_abstraction);

  auto frame_transformer = context_->getFrameManager()->getTransformer();
  rclcpp::Node::SharedPtr node = ros_node_abstraction->get_raw_node();
```

Reply:
Thanks, Mike, this's exactly what I did, sorry for not pushing my changes -- I'll fix the planning scene plugin and push them all together

Reply:
@JafarAbdi I think that making `node_` a class variable might end up being a mistake as it defeats the entire reason to use a WeakPtr in the first place. (Ref. https://www.learncpp.com/cpp-tutorial/15-7-circular-dependency-issues-with-stdshared_ptr-and-stdweak_ptr/) As best I can tell, the node_ is only used to initialize the PSM which I think could have it's own node and executor just like the tf_transformer. I can work on a PR to fix this

Currently, loading the PlanningSceneDisplay and then removing it causes rviz2 to crash with:
```
what():  RuntimeAssertionException: name must not be empty in destroySceneNode at /tmp/binarydeb/ros-eloquent-rviz-ogre-vendor-7.0.4/obj-x86_64-linux-gnu/ogre-v1.12.1-prefix/src/ogre-v1.12.1/OgreMain/src/OgreSceneManager.cpp (line 760)
```

Reply:
Thanks @mlautman , Does this [PR](https://github.com/ros-planning/moveit2/pull/168) fixes it .?